### PR TITLE
Add improvements for the EKS Discover UI.

### PR DESCRIPTION
**Pull request by AntonAM**
This PR adds a few small improvements to the Discover EKS flow - 
- Add a note informing user about access entries auth mode requirements
- Change label text for the manual instructions.
- If we encounter an error about agent already installed on the cluster, just skip to the step 2 "waiting" (there's small time window when this can happen).

### CODE REVIEW

File: web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx [hunk 1]
```diff
@@ -218,7 +218,12 @@ export function EnrollEksCluster(props: AgentStepProps) {
         emitErrorEvent(
           'unknown error: no results came back from enrolling the EKS cluster.'
         );
-      } else if (result.error) {
+      } else if (
+        result.error &&
+        !result.error.message.includes(
+          'teleport-kube-agent is already installed'
+        )
```
Comment (avatus) [inline web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx:225]: does this have any specific error code we could check for instead?

File: web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx [hunk 2]
```diff
@@ -293,6 +298,17 @@ export function EnrollEksCluster(props: AgentStepProps) {
       {fetchClustersAttempt.status === 'failed' && !hasIamPermError && (
         <Danger mt={3}>{fetchClustersAttempt.statusText}</Danger>
       )}
+      <Text mt={4} mb={-3}>
```
Comment (avatus) [inline web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx:301]: What is the purpose of the negative value here? are we doing it to try and not affect the positioning of something else?

File: web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx (hunk 1 above)
Comment (rudream) [reply web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx:225]: Checking for error code as Michael suggested is ideal, but if we're going to check for a substring in an error message, we should make sure to leave a note where the error message in defined in the backend saying that it needs to be kept in sync with this.

File: web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx [hunk 3]
```diff
@@ -346,7 +362,7 @@ export function EnrollEksCluster(props: AgentStepProps) {
                 }}
                 pl={0}
               >
-                Or click here to see instructions for manual enrollment
+                Or do manual enrollment
```
Comment (rudream) [inline web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx:365]:
```suggestion
                Or enroll manually
```
nit suggestion

File: web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx (hunk 1 above)
Comment (AntonAM) [reply web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx:225]: Not really, it's not an AWS error, just our internal message.

File: web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx (hunk 2 above)
Comment (AntonAM) [reply web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx:301]: I copied it from RDS discover where we use similar styled note, I think it's just to make margin to look better

Negative margin:
image

vs 0 margin

image

Not really sure which one looks better, I can remove negative margin if you think it's not needed.

Comment (avatus) [reply web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx:301]: its a nit at this point so feel free to ignore, but it'd probably make more sense to reduce the `mt` of the element right below it (the `Select the AWS Region...` Text element) than to use negative margin here.

**Review by avatus — APPROVED** [review]

File: web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx (hunk 1 above)
Comment (AntonAM) [reply web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx]: I've added a comment 👍

File: web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx (hunk 2 above)
Comment (AntonAM) [reply web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx]: Makes sense, changed the margins 👍

**Review by zmb3 — APPROVED** [review]

**Pull request merged**